### PR TITLE
Adding resource requests and limits and health probes

### PR DESCRIPTION
**Pull request by lieberlois**
This pull requests adds default resource limits and health probes to the operator and kube-rbac-proxy containers.

### CODE REVIEW

File: charts/opensearch-operator/values.yaml [hunk 1]
```diff
@@ -3,11 +3,32 @@ tolerations: []
...
   resources:
     limits:
-      cpu: 2
-      memory: 1Gi
+      cpu: 100m
+      memory: 300Mi
     requests:
-      cpu: 2
-      memory: 1Gi
+      cpu: 20m
```
Comment (swoehrl-mw) [inline charts/opensearch-operator/values.yaml:9]: Just 20m CPU is very low. If users need to squeeze out every bit they can lower it themselves but I think the default should be higher.
My suggestion for CPU: requests: 100m, limit 200m.
Maybe add a comment that 20m is a minimum.

**Review by swoehrl-mw — CHANGES_REQUESTED** [review]

File: charts/opensearch-operator/values.yaml [hunk 2]
```diff
@@ -3,11 +3,32 @@ tolerations: []
...
   resources:
     limits:
-      cpu: 2
-      memory: 1Gi
+      cpu: 200m
+      memory: 300Mi
     requests:
-      cpu: 2
-      memory: 1Gi
+      cpu: 100m
```
Comment (idanl21) [inline charts/opensearch-operator/values.yaml:9]: did you check that numbers? cause as i remember the operator is not working well with under than 1 cpu and 1gb mem

**Review by idanl21 — COMMENTED** [review]
Just needs to be sure about the numbers

File: charts/opensearch-operator/values.yaml (hunk 2 above)
Comment (lieberlois) [reply charts/opensearch-operator/values.yaml:9]: @idanl21 In our deployment, the Operator idles at usages of ~275 MB RAM and 10 Mi CPUs with spikes of ~370 MB RAM and 20 Mi CPUs. I raised the memory requests & limits a little bit to match that 👍

**Review by swoehrl-mw — APPROVED** [review]

**Review by prudhvigodithi — APPROVED** [review]

**Comment by prudhvigodithi** [comment]
Thanks @lieberlois, Hey @idanl21 if you are ok with this, lets merge it.

**Pull request merged**